core/lib/dal/src/blocks_dal.rs

```diff
@@ -1774,7 +1787,16 @@ impl BlocksDal<'_, '_> {
         let unsorted_factory_deps = self
             .get_l1_batch_factory_deps(L1BatchNumber(storage_batch.number as u32))
             .await?;
-        let header: L1BatchHeader = storage_batch.clone().into();
+
+        let l2_to_l1_logs = self
+            .get_l2_to_l1_logs_for_batch::<UserL2ToL1Log>(L1BatchNumber(
+                storage_batch.number as u32,
+            ))
+            .await?;
+
+        let header: L1BatchHeader = storage_batch
+            .clone()
+            .into_l1_batch_header_with_logs(l2_to_l1_logs);
         let Ok(metadata) = storage_batch.try_into() else {
```

Comment:
Nit: Makes sense to do this conversion before loading logs so that an additional DB query isn't required if the batch metadata is incomplete.

Reply:
sure.